wifiphisher.py

```diff
@@ -881,32 +884,127 @@ def get_hostapd():
+        if args.template and args.template in template_database:
+
+            # set the template name
+            template_name = args.template
+
+            # set the template selected to avoid interactive selection
+            template_selected = True
+
+        # loop until a valid template is selected
+        while not template_selected:
 
-        # if template is incomplete locally, delete and ask for a download
-        if not phishingpage.check_template(TEMPLATE_NAME):
+            # display start of template names
+            print "\nList Of All Templates: \n"
 
-            # clean up the previous download
-            phishingpage.clean_template(TEMPLATE_NAME)
+            # display the templates
+            for number in range(len(template_names)):
+                print ("[" + G + str(number + 1) + W + "] " +
+                       template_names[number])
+
+            # get user's choice
+            choosen_template = raw_input("Please Select One Of The Above "
+                                         "Templates: ")
+
+            # try to convert the input to integer
+            try:
+
+                template_number = int(choosen_template)
+
+            except ValueError:
+
+                # placed to avoid a program crash in case of non integer input
+                pass
+
+            if (type(template_number) is int and
+                    template_number in range(1, len(template_names) + 1)):
+
+                template_selected = True
+
+            else:
 
-            # get user's response
-            response = raw_input("Template is available online. Do you want"\
-            " to download it now? [y/n] ")
+                print "[" + R + "-" + W + "] Wrong Input! Please Try Again"
+
+            # remove 1 from template number which was added for display reasons
+            template_number -= 1
+
+            try:
+                # set the template name
+                template_name = template_names[template_number]
+            except IndexError:
+
+                # placed to avoid a program crash in case of unexpected
+                # template number
+                pass
+
+        # check to see if the template is local
+        if template_database[template_name] is None:
+
+            # in case the template is local break out of infinite loop
+            break
+
+        # in case the template is online
+        else:
+
+            # if template is incomplete locally, delete and ask for a download
+            if not phishingpage.check_template(template_name):
+
+                print ("[" + R + "-" + W + "] Deleting " + template_name +
+                       ": Template incomplete")
+
+                # clean up the previous download
+                phishingpage.clean_template(template_name)
+
+                # get user's response
+                response = raw_input("Template is available online. Do you "
+                                     "want to download it now? [y/n] ")
```

Comment:
Something doesn't seem right. This prompt appears to happen only when the template is incomplete locally, while it should also happen when the template exists only online (e.g. when the user runs the script for the first time).